charts/perses/templates/tests/test-config.yaml

```diff
@@ -0,0 +1,40 @@
+{{- if .Values.testFramework.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-test
+  namespace: {{ .Release.Namespace }}
+  labels:
+    type: helm-chart-test
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded"
+data:
+  run.sh: |-
+
+    #!/usr/bin/env bats
+    
+    load "/usr/lib/bats/bats-detik/utils"
+    load "/usr/lib/bats/bats-detik/detik"
+    
+    DETIK_CLIENT_NAME="kubectl"
+    
+    @test "Verify successful deployment and running status of the {{ .Release.Name }} pod" {
+        verify "there is 1 statefulset named '{{ include "perses.fullname" . }}'"
+        try "at most 3 times every 20s to get pods named '{{ include "perses.fullname" . }}.*' and verify that '.status.phase' is 'running'" 
```

Comment:
Generally I'd discourage usage of variables. Well I can understand that people disagree.
And indeed if the tests are executed at runtime and not at build time, it can be painful if we forget to update the hard coded values everytime. So in the end ok but what is your thought?

Reply:
Yes, I totally agree with you. It will be painful and I don't have a solution on how to get rid of hard-coded values. However, GitHub Actions workflow runs these Helm tests automatically on every pull request, providing quick feedback if any issues arise from changes. This helps catch potential problems before the PR is merged :)

Reply:
### Screenshot of the test suite result
<img width="1115" alt="image" src="https://github.com/user-attachments/assets/fbbadfd0-c105-4d16-a1c1-755a8c42c2d8">

Reply:
So it's creating a k8s environment in pipeline? cool ! Like with setup-envtest? https://pkg.go.dev/sigs.k8s.io/controller-runtime/tools/setup-envtest

Reply:
Yes, exactly. It's creating a [kind cluster](https://kind.sigs.k8s.io/docs/user/quick-start/) in the pipeline. :) 

[link to the GitHub Action workflow file](https://github.com/ibakshay/helm-charts/blob/fabd5710eda98df8f065a63d01ab0689991340c8/.github/workflows/lint-and-test.yaml#L19-L22)

### pipeline logs
![image](https://github.com/user-attachments/assets/6cb8e75b-e318-4620-8e83-81f768a07363)